### PR TITLE
Fix cc extractor (which references mtv extractor)

### PR DESCRIPTION
**Pull request by jfogelman**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Recent changes to cc.com causes failures for downloading videos. I added some additional checking and regex checks for the extractor to address these changes.

### CODE REVIEW

**Comment by nixxo** [comment]
https://github.com/ytdl-org/youtube-dl/pull/30149

**Comment by pukkandan** [comment]
@Sipherdrakon can u have a look at how this and the upstream PR interact with your partial mtv fix

File: yt_dlp/extractor/mtv.py [hunk 1]
```diff
@@ -320,7 +320,12 @@ def _extract_mgid(self, webpage):
             main_container = self._extract_child_with_type(data, 'MainContainer')
             ab_testing = self._extract_child_with_type(main_container, 'ABTesting')
             video_player = self._extract_child_with_type(ab_testing or main_container, 'VideoPlayer')
-            mgid = video_player['props']['media']['video']['config']['uri']
+            if video_player:
+                mgid = video_player['props']['media']['video']['config']['uri']
```
Comment (pukkandan) [inline yt_dlp/extractor/mtv.py:324]:
```suggestion
            mgid = try_get(video_player, lambda x: x['props']['media']['video']['config']['uri'])
```

**Comment by Sipherdrakon** [comment]
So I finally had time to check the changes made here and within youtube-dl. I checked every site attached to the mtvextractor and found that some parts of some of the sites were unsupported and that the PR upstream works far better than this PR, but it isn't perfect.

I'm not sure how to make changes to this PR but I can make a new one to fix the upstream PR to work with all supported URLs that were tested.

**Comment by pukkandan** [comment]
@Sipherdrakon You can open a new PR

**Pull request closed without merging**